contrib/scripts/kind.sh

```diff
@@ -49,6 +49,11 @@ reg_name="kind-registry"
 reg_port="5000"
 running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
 if [[ "${running}" != "true" ]]; then
+  while ! docker pull registry:2
```

Comment:
Forever?

Reply:
i figured we might as well retry forever, the workflow has a timeout so it'll fail at some point https://github.com/cilium/cilium/blob/master/.github/workflows/conformance-datapath.yaml#L119

but yeah maybe giving up after something like 10 failures also seems reasonable 👀

Reply:
This script is also used in local development environments, though admittedly I've never seen this class of failure there. Still, would be nice to bound that and give a nice message to developers.

Reply:
let's retry 10 times and fail 🐳 i'll update the PR and mark it ready for review

Reply:
😇 
```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
docker pull registry:2 failed. Sleeping for 1 second and trying again...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
ERROR: 'docker pull registry:2' failed 10 times. Please make sure docker is running
```